modules/package.nix

```diff
@@ -301,6 +301,11 @@ in {
       type = either path package;
       default = pkgs.fetchurl { url = "mirror://hackage/${config.name}.tar.gz"; inherit (config) sha256; };
       defaultText = "pkgs.fetchurl { url = \"mirror://hackage/\${config.name}.tar.gz\"; inherit (config) sha256; };";
+      # Make sure paths have a context so they will be included in the derivation
+      # inputs for the component derivations.  Without this sandbox builds fail
+      # cannot see the input and fail with the error:
+      #   do not know how to unpack source archive /nix/store/...
+      apply = v: if isString v && __getContext v == {} then __appendContext v { ${v} = { path = true; }; } else v;
```

Comment:
I still have some doubts about using `appendContext`. It does seems to work but also it seems to do (in this case at least) the same things that `storePath` does, which is not allowed in pure mode. We don't want to accidentaly rely on a nix bug :joy:

Reply:
That said, the type of `src` is `either path package` according to the nixos documentation `A filesystem path is anything that starts with a slash when coerced to a string`, so path is not necessarily a store path.
In case path is not a store path, `appendContext` would fail.
```
nix-repl> __appendContext "" { "/home/andrea" = { path = true; }; } 
error:
       … while calling the 'appendContext' builtin

         at «string»:1:1:

            1| __appendContext "" { "/home/andrea" = { path = true; }; }
             | ^

       error: context key '"/home/andrea"' is not a store path

       at «string»:1:22:

            1| __appendContext "" { "/home/andrea" = { path = true; }; }
             |                      ^
```
Perhaps we need to be more careful with this? Maybe we can have a `storePath` type? or move the call to  `appendContext` elsewhere?